### PR TITLE
lib: Add CRUD support for HelmRelease,GitRepository

### PR DESCRIPTION
**Pull request by surajssd**

### CODE REVIEW

File: lib/source-controller/lib.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type GitRepoConfig struct {
+	c          client.Client
+	kubeconfig []byte
+	gr         *api.GitRepository
+}
+
+type gitRespoConfigOpt func(*GitRepoConfig)
```
Comment (iaguis) [inline lib/source-controller/lib.go]:
```suggestion
type gitRepoConfigOpt func(*GitRepoConfig)
```

**Comment by surajssd** [comment]
@knrt10 @iaguis can we merge this one so as to close https://github.com/kinvolk/lokomotive/issues/1550 ?

**Comment by iaguis** [comment]
Let me do a quick review.

File: lib/client.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package lib
```
Comment (iaguis) [inline lib/client.go:1]: Can we add a license header?

File: lib/client.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func getRestClient(kubeconfig []byte) (*restclient.Config, error) {
+	if len(kubeconfig) == 0 {
+		client, err := clientcmd.BuildConfigFromFlags("", "")
+		if err != nil {
+			return nil, fmt.Errorf("neither kubeconfig not provided nor in-cluster config available: %w", err)
```
Comment (iaguis) [inline lib/client.go]:
```suggestion
			return nil, fmt.Errorf("kubeconfig not provided and in-cluster config not available: %w", err)
```

File: lib/client.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+		}
+
+		return client, nil
+	}
+
+	clientConfig, err := clientcmd.NewClientConfigFromBytes(kubeconfig)
+	if err != nil {
+		return nil, fmt.Errorf("creating client config failed: %w", err)
```
Comment (iaguis) [inline lib/client.go]:
```suggestion
		return nil, fmt.Errorf("creating client config: %w", err)
```

File: lib/client.go [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+	}
+
+	restConfig, err := clientConfig.ClientConfig()
+	if err != nil {
+		return nil, fmt.Errorf("converting client config to rest client config failed: %w", err)
```
Comment (iaguis) [inline lib/client.go]:
```suggestion
		return nil, fmt.Errorf("converting client config to rest client config: %w", err)
```

File: lib/client.go [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	}
+
+	return restConfig, nil
+}
+
+func GetKubernetesClient(kubeconfig []byte, s *runtime.Scheme) (client.Client, error) {
+	restConfig, err := getRestClient(kubeconfig)
+	if err != nil {
+		return nil, fmt.Errorf("creating a REST client: %w", err)
```
Comment (iaguis) [inline lib/client.go]:
```suggestion
		return nil, fmt.Errorf("creating REST client: %w", err)
```

File: lib/doc.go [hunk 7]
```diff
@@ -0,0 +1 @@
+package lib
```
Comment (iaguis) [inline lib/doc.go:1]: Ditto about license header

File: lib/helm-release/doc.go [hunk 8]
```diff
@@ -0,0 +1 @@
+package helmrelease
```
Comment (iaguis) [inline lib/helm-release/doc.go:1]: Ditto about license header

File: lib/helm-release/lib.go [hunk 9]
```diff
@@ -0,0 +1,113 @@
+package helmrelease
```
Comment (iaguis) [inline lib/helm-release/lib.go:1]: Ditto about license header

File: lib/source-controller/doc.go [hunk 10]
```diff
@@ -0,0 +1 @@
+package sourcecontroller
```
Comment (iaguis) [inline lib/source-controller/doc.go]: Ditto about license header

File: lib/source-controller/lib.go [hunk 11]
```diff
@@ -0,0 +1,113 @@
+package sourcecontroller
```
Comment (iaguis) [inline lib/source-controller/lib.go]: Ditto about... well, I think you get the idea 😂

File: lib/helm-release/lib.go [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+	kubeconfig []byte
+}
+
+type helmReleaseConfigOpt func(*HelmReleaseConfig)
+
+var scheme *runtime.Scheme
+
+func init() {
+	scheme = runtime.NewScheme()
+	_ = api.AddToScheme(scheme)
```
Comment (iaguis) [inline lib/helm-release/lib.go:26]: Do we need an init() function? I heard they're evil in libraries

**Review by iaguis — CHANGES_REQUESTED** [review]

File: lib/helm-release/lib.go (hunk 12 above)
Comment (surajssd) [reply lib/helm-release/lib.go:26]: We need this to register those APIs, so as to talk to k8s api-server.

**Review by iaguis — APPROVED** [review]
lgtm

**Pull request merged**